packages/plugin-docusaurus/src/client/theme/SearchBar/index.tsx

```diff
@@ -64,42 +65,64 @@ function snippet(item: Hit): JSX.Element {
 export default function SearchBar(): JSX.Element {
   const isBrowser = useIsBrowser()
   const { siteConfig } = useDocusaurusContext()
+  const location = useLocation()
   const containerRef = useRef(null)
   const { colorMode } = useColorMode()
-  const indexData = usePluginData(PLUGIN_NAME)
+  const { searchData, versions } = usePluginData(PLUGIN_NAME) as PluginData
+  const [database, setDatabase] = useState<OramaWithHighlight>()
+  const searchBaseUrl = useBaseUrl(INDEX_FILE)
+
+  const version = useMemo(() => {
+    if (!isBrowser) {
+      return undefined
+    }
+
+    const pathname = location.pathname
+
+    for (const v of versions) {
+      if (pathname.startsWith(v.path)) {
+        return v.name
+      }
+    }
+
+    return versions[0].name
+  }, [isBrowser, versions, location])
 
   useEffect(() => {
-    if (!containerRef.current || !isBrowser) {
+    if (!containerRef.current || !isBrowser || !database) {
       return undefined
     }
 
     const search = autocomplete({
+      placeholder: 'Search ...',
       container: containerRef.current,
       // @ts-expect-error render typing here is for preact, react also works
       renderer: { createElement, Fragment, render },
       openOnFocus: true,
       detachedMediaQuery: '', // always detached
-      async getSources({ query }): Promise<any> {
-        const orama = await getOrama(siteConfig.baseUrl, indexData)
-
+      async getSources({ query: term }): Promise<any> {
         return [
           {
             sourceId: 'orama',
             async getItems() {
-              const results = await orama(query)
+              const results = await searchWithHighlight(database, {
+                term,
+                properties: ['sectionTitle', 'sectionContent', 'type']
```

Comment:
Would you mind searching only on these properties? Should we allow customization?

Reply:
Right now the properties come from the parse-doc plugin, so they are fixed. We can eventually revisit in the future.